packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -97,6 +110,55 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
     return null;
   }
 
+  const handleTestQuery = async () => {
+    setTestQueryResults(null);
+    setQueryError(null);
+
+    const value: ExposureQuery = {
+      name: exposureQuery.name,
```

Comment:
`exposureQuery` is only defined when editing an existing one, not when adding a new one.